File: application-server/src/main/java/dto/common/ExceptionDto.java
package dto.common;

import dto.common.type.ErrorCode;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
public class ExceptionDto {

    private final int code;
    private final String message;

    public static ExceptionDto from(final ErrorCode errorCode) {
        return new ExceptionDto(errorCode.getCode(), errorCode.getMessage());
    }

    public static ExceptionDto of(final int code, final String message) {
        return new ExceptionDto(code, message);
    }
}

File: anomaly/detector/core/dto/common/exception_dto.py
from core.dto.common.error_code import ErrorCode


class ExceptionDto:
    def __init__(self, error_code: ErrorCode):
        self.status: int = error_code.value
        self.message: str = error_code.name

    @staticmethod
    def of(error_code: ErrorCode):
        return ExceptionDto(error_code)
